posts/2024-07-30-24.0.0.8-beta.adoc

```diff
@@ -0,0 +1,816 @@
+
+
+With the new MicroProfile OpenAPI 4.0 feature (`mpOpenAPI-4.0`), documentation is now produced in link:https://spec.openapis.org/oas/v3.1.0.html[OpenAPI 3.1 format], updated from 3.0 in previous versions. Changes in OpenAPI 3.1 include:
+
+* Use of full JSON Schema 2020-12 draft for data object schemas (updated from a subset of an older JSON schema draft in 3.0)
+* Support for documenting webhooks
+* Reusable PathItems
+* Updates to the model API so that it directly reflects the OpenAPI 3.1 format
+* Additions to the annotations API to allow users to take advantage of the new features of OpenAPI 3.1
+
+A detailed list of changes can be found in the link:https://download.eclipse.org/microprofile/microprofile-open-api-4.0-RC4/microprofile-openapi-spec-4.0-RC4.html#release_notes_40[release notes of the specification].
+
+This is an early beta release to support the ratification of MicroProfile 7.0. As such, some function which was available in previous versions of the feature are not yet implemented:
+
+- Documenting more than one web module as link:https://openliberty.io/docs/latest/documentation-openapi.html#multi-module[can be configured in previous versions]
+- Validation of the produced OpenAPI document
+
+=== Further resources:
+
+For more information, see the following resources:
+
+- link:https://download.eclipse.org/microprofile/microprofile-open-api-4.0-RC4/microprofile-openapi-spec-4.0-RC4.html[MP OpenAPI 4.0 RC4 specification document] and link:https://download.eclipse.org/microprofile/microprofile-open-api-4.0-RC4/apidocs/[API Javadoc].
+- link:https://openliberty.io/docs/latest/documentation-openapi.html[API documentation with OpenAPI]
+- Guide: link:https://openliberty.io/guides/microprofile-openapi.html[Documenting RESTful APIs]
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28869
+// Contact/Reviewer: KyleAure,njr-11
+// // // // // // // //
+
+[#data]
+== Bring your own Jakarta Data provider with Jakarta Data API 1.0
+
+Jakarta Data 1.0 is a new Jakarta EE specification that standardizes a repository-based programming model for data access across relational and non-relational databases.
+
+The Jakarta Data API feature (`dataContainer-1.0`) enables you to use third-party Jakarta Data providers in Open Liberty without including the built-in provider for EclipseLink. This configuration is useful if you want to use a different Jakarta Persistence-based Jakarta Data provider, such as Hibernate, without colliding on the Jakarta Persistence Entity annotation. It's also useful if you want to use Jakarta NoSQL exclusively and do not want the overhead of the built-in provider for relational databases.

```

Comment:
I prefer "Jakarta Data API" because it gives the user a better impression of what it is. A user will not know what a "Jakarta Data Container" is. It is not a concept from the spec, nor is it intuitive. But they will know exactly what they are getting with "Jakarta Data API".

Reply:
The main concern was the inconsistency with the other xxxContainer features discussed in the blog since both faces and persistence Container features are mentioned.